### PR TITLE
cereal: add cereal a header-only c++11 serialization library

### PR DESCRIPTION
**Pull request by miartad**
cereal is a c++11 serialization library licensed under the BSD license.

I'm not owner or maintainer of said library and not sure whether approval of those is necessary for adding them to meta-oe, however I do use cereal in my embedded projects and thought it might be useful for others as well, hence providing my bitbake recipe.

### CODE REVIEW

File: meta-oe/recipes-devtools/cereal/cereal_1.0.bb [hunk 1]
```diff
@@ -24,3 +26,11 @@ EXTRA_OECMAKE += "\
 "
 
 S = "${WORKDIR}/git"
+
+PROVIDES += "${PN}-dev"
```
Comment (miartad) [inline meta-oe/recipes-devtools/cereal/cereal_1.0.bb:30]: cereal provides only header files for development, no binary or library is created during build. hence reflect that by providing dev package

**Comment by kraj** [comment]
this is now applied via e1b438c138

**Comment by kraj** [comment]
We already have meta-oe/recipes-support/libcereal/libcereal_1.3.0.bb

I recommend ./meta-oe/recipes-devtools/cereal/cereal_1.0.bb is removed
again, and any relevant changes added to libcereal instead.

**Comment by miartad** [comment]
@kraj you are correct, somehow I missed that. I plan to make and push the changes this week.

**Comment by kraj** [comment]
> @kraj you are correct, somehow I missed that. I plan to make and push the changes this week.

I am still expecting you to consolidate this, otherwise, let me know, I will have to revert this pull.

**Comment by miartad** [comment]
@kraj updated to use already existing libcereal recipe

**Comment by kraj** [comment]
> @kraj updated to use already existing libcereal recipe

I thought one of the recipes could be deleted as well isnt it so ?

**Comment by miartad** [comment]
rebased to master branch

**Comment by kraj** [comment]
this has been merged in master

**Pull request closed without merging**